spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/token/authentication/XsuaaAudienceValidator.java

```diff
@@ -42,6 +45,7 @@ public OAuth2TokenValidatorResult validate(Jwt token) {
 						"Missing audience " + xsuaaServiceConfiguration.getAppId(), null));
 			}
 		}
+		return OAuth2TokenValidatorResult.success();
```

Comment:
I don't like the idea of returning this as default if the logic above fails. Lets keep the logic for application & broker plan in one validator, but make the validator configurable.

Reply:
This is the compromise of not providing a variant of validators, which is compatible then... 
In case you dislike this approach, as well as the first one, the affected customers need to copy a lot of logic, which is already available and can be easily configured in case of having two separate validators.

Reply:
Done